### PR TITLE
Uar 1587 add nocs review update page no change

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1587

### Change description
Update the 'Review' page to include the new NOCs for a no change update/remove journey.

- Agreed with Vernon - If feature flag is active then we still want to show the invalid NOC (member of a firm) as the user will not have a chance to remove it during no-change and they should be aware of the current state of their data before submitting.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/includes/check-your-answers/beneficial-owner-gov.html [hunk 1]
```diff
@@ -31,38 +31,44 @@
...
-    {% set boNOCOwnerOfLandPersonJurisdictionsArray = bog.owner_of_land_person_nature_of_control_jurisdictions %}
-    {% include "includes/check-your-answers/nocs/beneficial-owner-owner-of-land-person-noc.html" %}
-  {% endset %}
-
-  {% set bogOwnerOfLandOtherEntityNOCs %}
-    {% set boNOCOwnerOfLandOtherEntityJurisdictionsArray = bog.owner_of_land_other_entity_nature_of_control_jurisdictions %}
-    {% include "includes/check-your-answers/nocs/beneficial-owner-owner-of-land-other-entity-noc.html" %}
-  {% endset %}
-{% endif %}
+{% set bogBeneficialOwnerNonLegalFirmNOCs %}
```
Comment (mattch1) [inline views/includes/check-your-answers/beneficial-owner-gov.html:34]: Does this mean we'll be showing the new nocs even when the feature flag is false?

Comment (markpit) [reply views/includes/check-your-answers/beneficial-owner-gov.html:34]: no, the new Nocs will be hidden when flag is false. These sections are just to build the html if a noc is present. For the no change journey, the old invalid noc is to be displayed if present in CHS so they could potentially have all 7 displayed for a no change journey. The page will construct the html depending on what nocs they tick so we can just let them do their job and then control what is displayed further down when we join the noc html together to display using the boNOCHtml variable.

File: test/controllers/update/update.review.statement.controller.spec.ts [hunk 2]
```diff
@@ -261,6 +263,52 @@ describe("Update review overseas entity information controller tests", () => {
...
     });
 
+    test.each([
+      ["on remove journey", APPLICATION_DATA_REMOVE_BO_MOCK],
+      ["on update journey", APPLICATION_DATA_UPDATE_BO_MOCK]
+    ])(`renders the ${UPDATE_REVIEW_STATEMENT_PAGE} page with the NOCs when flag FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC on, %s`, async (_journeyType, mockAppData) => {
+      mockIsActiveFeature.mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_TRUSTS_WEB
+      mockIsActiveFeature.mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC
+
+      mockGetApplicationData.mockReturnValue({
```
Comment (lduranteau) [inline test/controllers/update/update.review.statement.controller.spec.ts:273]: where is it defined that we are in a noChange journey?

File: test/controllers/update/update.review.statement.controller.spec.ts [hunk 3]
```diff
@@ -261,6 +263,52 @@ describe("Update review overseas entity information controller tests", () => {
...
+    });
+
+    test.each([
+      ["on remove journey", APPLICATION_DATA_REMOVE_BO_MOCK],
+      ["on update journey", APPLICATION_DATA_UPDATE_BO_MOCK]
+    ])(`renders the ${UPDATE_REVIEW_STATEMENT_PAGE} page with the NOCs when flag FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC off, %s`, async (_journeyType, mockAppData) => {
+      mockIsActiveFeature.mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_TRUSTS_WEB
+      mockIsActiveFeature.mockReturnValueOnce(false); // FEATURE_FLAG_ENABLE_PROPERTY_OR_LAND_OWNER_NOC
+
+      mockGetApplicationData.mockReturnValue({
```
Comment (lduranteau) [inline test/controllers/update/update.review.statement.controller.spec.ts:296]: where is it defined that we are in a noChange journey?

File: views/includes/check-your-answers/beneficial-owner-gov.html (hunk 1 above)
Comment (mattch1) [reply views/includes/check-your-answers/beneficial-owner-gov.html:34]: Yes I see that now.

**Review by mattch1 — APPROVED** [review]
LGTM

**Review by mwestacott — APPROVED** [review]

**Pull request merged**